### ISSUE TITLE
Experiencing unknown blob error on push

### ISSUE THREAD

**Issue by gudleik** [docker/build-push-action#38]
I have a workflow containing multiple jobs building and pushing images to Github Packages.
This works great most of the time, but sometimes a random job is failing during docker push with an `unknown blob` error.

The jobs are declared like this:

```yaml
  build-maintenance:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: docker/build-push-action@v1
        with:
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}
          registry: docker.pkg.github.com
          repository: XXX/YYY/maintenance
          path: maintenance
          target: production
          tag_with_sha: true
```

Tail of job output:

```
Successfully built c6ac9d5faeba
Successfully tagged docker.pkg.github.com/XXX/YYY/maintenance:sha-eac817d
Pushing image [docker.pkg.github.com/XXX/YYY/maintenance:sha-eac817d]
The push refers to repository [docker.pkg.github.com/XXX/YYY/maintenance]
1e64616ff473: Preparing
02e9da9d44c5: Preparing
3810cc0c140f: Preparing
3e207b409db3: Preparing
3e207b409db3: Layer already exists
3810cc0c140f: Layer already exists
unknown blob
Error: exit status 1
exit status 1
Usage:
  github-actions build-push [flags]

Flags:
  -h, --help   help for build-push
```

**Comment by prise6** [comment]
Any news on this 
Got same issue. I wonder if this linked to limited free space for public projet ?

**Comment by gudleik** [comment]
I tried building and pushing without using this package, and experienced same errors.
When I changed from using short prefixed sha (sha-abcdefg) to full sha as image tags, the problem disappeared.

Maybe you can make tagging a little more customizable, like
```yaml

# use full sha
uses: docker/build-push-action@v1
with:
  tag: ${{GITHUB_SHA}}

# prefixed short sha like today
uses: docker/build-push-action@v1
with:
  tag: sha-${{GITHUB_SHA::7}}
```

**Comment by derTobsch** [comment]
@leoschweizer same here. We also use the `latest` tag see https://github.com/synyx/urlaubsverwaltung/runs/679001614#step:6:26

**Comment by RichiCoder1** [comment]
ditto. Seeing this pretty consistently.

**Comment by wujood** [comment]
I have the same issue, but it seems that if you retry it a couple of times it sometimes works. This seems to be a general issue?

**Comment by crazy-max** [comment]
Don't think this is related to this action but an issue with the GitHub Package Registry itself. You can get support about GitHub Packages [here](https://support.github.com/contact/feedback?contact%5Bcategory%5D=github-packages). Thanks.

**Comment by bhordupur** [comment]
> I have the same issue, but it seems that if you retry it a couple of times it sometimes works. This seems to be a general issue?

We have got the same issue last from yesterday as well. After trying 2/3 times it works. And sometimes even on the first try, it works.

**Comment by crazy-max** [comment]
@bhordupur https://github.com/docker/build-push-action/issues/148#issuecomment-702278002

**Comment by bhordupur** [comment]
Some time ago, we have got this fixed. It was the issue from the Azure side though.

**Comment by lupusllc** [comment]
@bhordupur we've been having this happen intermittently for quite some time to Azure Container Registries, what was the issue they/you found on the Azure side?

**Comment by bhordupur** [comment]
We have had geo-replication enabled for the ACR registry. We needed to disable the routing to the edge location which seemed to fix the issue. After a week if I remember correctly we enabled it. The issue is also documented on the Azure side. Here is the [link](https://gitlab.com/relex-work/intradayplanning/-/merge_requests/602).

Before that, we tried to delete the tag from the remote ACR and pushed the same image and it was more of luck. In some pushes, it worked without issues and others did not. @lupusllc
